Add rif tests

BonfaceKilz opened the pull request:
### Description

This PR introduces RDF unit test cases for the RIF fetches.  For more context read [gn3/ADR-000](https://issues.genenetwork.org/topics/ADR/gn3/000-add-test-cases-for-rdf).

### Testing

- Make sure you have a local virtuoso server running.
- Make sure you set the following in your secrets file:
 ```
SPARQL_USER = "XXX"
SPARQL_PASSWORD = "XXXXX"
SPARQL_AUTH_URI="http://localhost:9082/sparql-auth/"
SPARQL_CRUD_AUTH_URI="http://localhost:9082/sparql-graph-crud-auth"
 ```
 - Run tests by: `pylint -k rdf`
 
 TODO:
- Update the CD linux container with above params.

BonfaceKilz committed:
Remove un-necessary triple in comment history RDF query.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Alphabetically sort the categories and pubmed_ids array.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Remove dead variable from RDF query.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Use correct modifier when getting species name in SPARQL.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Add RDF test cases.

* pytest.ini: Add rdf pytest marker.
* tests/test_data/ttl-files/test-data.ttl: New file with turtle test
data for RIF editing.
* tests/unit/db/rdf/test_wiki.py (test_sanitize_result): New test
case.
(test_get_comment_history): Ditto.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Use test fixtures to set-up a test named graph.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Use correct object modifier in triple when fetching comment history.

* gn3/db/rdf/wiki.py (get_comment_history) [query]: Replace
?symbolName with ?symbolName.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Update docstring for "update_wiki_comment".

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Use correct object modifier for a label when inserting a comment.

* gn3/db/rdf/wiki.py (update_wiki_comment) [query]:  Replace
"rdf:label" with "rdfs:label".

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Update docs to highlight sparql conf params in SECRETS file.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Make sure the the test graph is empty before running rdf tests.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Remove dead comment.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

jnduli commented on tests/fixtures/rdf.py:
```diff
@@ -0,0 +1,46 @@
+def get_sparql_auth_conf() -> dict:
+    """Fetch SPARQL auth configuration for the GN3_SECRETS file."""
+    sparql_conf = config.Config("")
+    sparql_conf.from_envvar("GN3_SECRETS")
```
I think these should be reorderd i.e. GN3_CONF takes precedence over GN3_SECRETS

jnduli commented on gn3/db/rdf/wiki.py:
```diff
@@ -43,6 +43,7 @@ def __sanitize_result(result: dict) -> dict:
     categories = result.get("categories")
```
minor: we could improve this by:
```
categories = result.get("categories", [])
if isinstance(categories, str):
  result["categories"] = [categories]
```

jnduli commented on README.md:
````diff
@@ -126,11 +126,24 @@ These configurations should be set in an external config file, pointed to with t
+FAHAMU_AUTH_TOKEN="XXXXXX"
+```
+
+Note: The sparql configurations are important for running tests I.e. `pytest -k rdf`.
````
Add a link to https://issues.genenetwork.org/topics/systems/virtuoso to help set up virtuoso

jnduli commented on README.md:
````diff
@@ -126,11 +126,24 @@ These configurations should be set in an external config file, pointed to with t
+```
+SPARQL_USER = "XXX"
+SPARQL_PASSWORD = "XXXXXXXXXX"
+SPARQL_AUTH_URI="http://localhost:9082/sparql-auth/"
````
can we change this to the default port virtuoso provides?

BonfaceKilz committed:
Load GN3_SECRETS after loading GN3_CONF settings.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Remove one-line if/else statement.

* gn3/db/rdf/wiki.py (__sanitize_result): When fetching pubmed_ids and
categories, default to [] if the key misses, and remove the subsequent
one-line if/else statement.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Update docs on how to use SPARQL default ports.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Update docs to use default virtuoso ports.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Add section on setting up virtuoso locally.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

Alexanderlacuna commented:
LGTM!

BonfaceKilz committed:
Revert "Remove one-line if/else statement."

This reverts commit 0d15840913c9cc401d1e74295aa5e17e34f2987b.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>

BonfaceKilz committed:
Fix or ignore pylint errors.

Signed-off-by: Munyoki Kilyungi <me@bonfacemunyoki.com>
